connect/runtime/src/test/java/org/apache/kafka/connect/integration/ExampleConnectIntegrationTest.java

```diff
@@ -159,4 +207,41 @@ private boolean checkForPartitionAssignment() {
             return false;
         }
     }
+
+    /**
+     * Confirm that the requested number of workers is up and running.
+     *
+     * @param numWorkers the number of online workers
+     * @return true if at least {@code numWorkers} are up; false otherwise
+     */
+    private Optional<Boolean> assertWorkersUp(int numWorkers) {
+        try {
+            int numUp = connect.activeWorkers().size();
+            return Optional.of(numUp >= numWorkers);
+        } catch (Exception e) {
+            log.error("Could not check active workers.", e);
+            return Optional.empty();
+        }
+    }
```

Comment:
Seems like these might be useful in all integration tests that use this framework. Does it make sense to add them (or something similar) to the `EmbeddedConnectCluster` class? One, it helps emphasize that `EmbeddedConnectCluster` is really just for testing purposes and is not intended for embedding in production applications. Second, it allows other integration tests to much more easily do what this test class is doing by waiting until Connect gets to some desirable state.

Reply:
This thought crossed my mind too. But I think I'd prefer to move incrementally with what we can generalize and bring into `EmbeddedConnectCluster`. But totally agree. So far I've kept all the assertions in the test classes because I haven't conclusively decided on a pattern.